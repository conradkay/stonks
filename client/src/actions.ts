import { Action, Stonk, Variant } from './exports'
type AddCartAction = { type: 'ADD_TO_CART'; id: number; amount: number; stonk: Stonk }
export const addToCart = (stonk: Stonk, amount: number): AddCartAction => {
  const id: number = Math.floor((Math.random() * 100000000000))
  return { type: 'ADD_TO_CART', stonk, id, amount }
}
export const buyStonks = (): Action => ({ type: 'BUY_STONKS' })
export const sellStonks = (): Action => ({ type: 'SELL_STONKS' })
export const changeAmountShop = (amount: number, index: number): Action => ({
  type: 'CHANGE_AMOUNT_SHOP', amount, index
})
export const changeAmountCart = (amount: number, index: number): Action => ({
  type: 'CHANGE_AMOUNT_CART', amount, index
})
export const changeAmountInventory = (amount: number, index: number): Action => ({
  type: 'CHANGE_AMOUNT_INVENTORY', amount, index
})
export const unSelectAllInventory = (): Action => ({
  type: 'UNSELECT_ALL_INVENTORY'
})
export const selectAllInventory = (): Action => ({
  type: 'SELECT_ALL_INVENTORY'
})
export const selectStonkInInventory = (index: number): Action => ({
  type: 'SELECT_STONK_IN_INVENTORY',
  index: index
})
export const selectAllCart = (): Action => ({ type: 'SELECT_ALL_CART' })
export const unSelectAllCart = (): Action => ({ type: 'UNSELECT_ALL_CART' })
export const selectStonkInCart = (index: number): Action => ({ type: 'SELECT_STONK_IN_CART', index: index })
export const removeStonk = (index: number): Action => ({ type: 'REMOVE_STONK', index: index })

export const openSnackbar = (message: string, variant: Variant): Action => ({
  type: 'OPEN_SNACKBAR', message, variant
})

export const closeSnackbar = (): Action => ({ type: 'CLOSE_SNACKBAR' })
export const sellStonk = (): Action => ({ type: 'SELL_STONKS' })
