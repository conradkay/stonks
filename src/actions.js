export const selectStonkInInventory = (index) => ({
    type: 'SELECT_STONK_IN_INVENTORY',
    index: index
});
export const setShopValue = (index, amount) => ({
    type: 'SET_VALUE_IN_SHOP', index, number: amount
});
export const sellStonks = () => ({
    type: 'SELL_STONKS'
});
export const unSelectAllInventory = () => ({
    type: 'UNSELECT_ALL_INVENTORY'
});
export const selectAllInventory = () => ({
    type: 'SELECT_ALL_INVENTORY'
});
export const addToCart = (stonk, amount) => {
    const largestId = Math.floor((Math.random() * 10000000));
    return {
        type: 'ADD_TO_CART',
        stonk,
        id: largestId,
        amount
    };
};
export const changeStonkAmountInInventory = (amount, index) => ({
    type: 'CHANGE_STONK_AMOUNT_IN_INVENTORY', index, amount
});
export const changeAmount = (amount, index) => ({
    type: 'CHANGE_AMOUNT', amount, index
});
export const removeStonk = (index) => ({ type: 'REMOVE_STONK', index: index });
export const selectAllCart = () => ({ type: 'SELECT_ALL_CART' });
export const unSelectAllCart = () => ({ type: 'UNSELECT_ALL_CART' });
export const buyStonks = () => ({ type: 'BUY_STONKS' });
export const selectStonkInCart = (index) => ({
    type: 'SELECT_STONK_IN_CART',
    index: index
});
export const changeFilter = (newFilter) => ({ type: 'CHANGE_FILTERS', newFilter });
export const openSnackbar = (message, variant) => ({
    type: 'OPEN_SNACKBAR',
    message,
    variant
});
export const closeSnackbar = () => ({ type: 'CLOSE_SNACKBAR' });
export const sellStonk = () => ({ type: 'SELL_STONKS' });
//# sourceMappingURL=actions.js.map