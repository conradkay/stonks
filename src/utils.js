import { stonks } from './exports';
export const categories = [
    'food', 'realEstate', 'office', 'service'
];
export const itemBetweenIndexes = (parent, index, newItem) => {
    return [...parent.slice(0, index), newItem, ...parent.slice(index + 1)];
};
export const getClassSnackbarVariant = (variant, classes) => {
    if (variant === 'success') {
        return classes.success;
    }
    else if (variant === 'warning') {
        return classes.warning;
    }
    else if (variant === 'error') {
        return classes.error;
    }
    else {
        return classes.error;
    }
};
export const defaultState = {
    amountStonksShop: [
        1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1,
        1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1
    ],
    allSelected: false,
    cart: stonks,
    inventory: [...stonks],
    activeFilters: [],
    money: 1000,
    snackbar: { open: false, message: 'dont show', variant: 'success' },
};
//# sourceMappingURL=utils.js.map