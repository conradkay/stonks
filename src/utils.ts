import { State, Variant, stonks, Stonk } from './exports'
/* used in header component as such and
*  <Link style={getStylesHeader(currComp === 'contact')} />
*/
export const getStylesHeader = (conditionTrue: boolean): { textDecoration: 'underline' } | {} => {
  if (conditionTrue) {
    return { textDecoration: 'underline' }
  }
  else {
    return {}
  }
}
// used as parent: [...itemBetweenIndexes(state.parent, 3, {})]
export const itemBetweenIndexes = (parent: Array<any>, index: number, newItem: any) => {
  return [...parent.slice(0, index), newItem, ...parent.slice(index + 1)]
}

export const getClassSnackbarVariant = (variant: Variant, classes: { success: string; warning: string; error: string }) => {
  // function to bypass rules on typescript [] notation, used in snackbarWrap component
  if (variant === 'success') {
    return classes.success
  }
  else if (variant === 'warning') {
    return classes.warning
  }
  else if (variant === 'error') {
    return classes.error
  }
  else {
    return classes.error
  }
}
export const generateOnes = (amount: number): number[] => { // generates some ones to put into state.shopAmounts
  const arr = [1]
  for (let i = 0; i < amount - 1; i++) {
    arr.push(1)
  }
  return arr
}
export const defaultState: State = {
  amountStonksShop: generateOnes(25),
  cart: [
    ...stonks.slice(0, -1),
    { ...stonks[stonks.length - 1], selected: true }
  ],
  inventory: [
    ...stonks.slice(0, -1),
    { ...stonks[stonks.length - 1], selected: true }
  ],
  activeFilters: [],
  money: 1000,
  snackbar: { open: false, message: 'dont show', variant: 'success' },
}
export const exampleStonk: Stonk = {
  amount: 2,
  id: 1,
  description: `floopderop`,
  name: 'floop',
  price: 101,
  pathToImage: 'https://',
  sellAmount: 1,
  selected: false
}
