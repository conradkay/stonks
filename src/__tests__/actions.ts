import * as actions from '../actions'
import { exampleStonk } from '../utils'
describe('core actions', () => {
  it('add to cart action', () => {
    expect(actions.addToCart(exampleStonk, 2).id.toString().length).toBeGreaterThanOrEqual(9)
    expect({ ...actions.addToCart(exampleStonk, 2), id: 3 }).toMatchSnapshot()
  })
  it('buy stonks action', () => expect(actions.buyStonks()).toMatchSnapshot())
  it('sell stonks action', () => expect(actions.sellStonks).toMatchSnapshot())
  it('change amount shop action', () => expect(actions.changeAmountShop(3, 2)).toMatchSnapshot())
  it('change amount cart action', () => expect(actions.changeAmountCart(3, 20)).toMatchSnapshot())
  it('change amount inventory action', () => expect(actions.changeAmountInventory(4, 40)).toMatchSnapshot())
  it('unSelectAllInventory/unSelectAllCart', () => {
    expect(actions.unSelectAllInventory()).toMatchSnapshot()
    expect(actions.selectAllCart()).toMatchSnapshot()
  })
  it('selectAllCart/selectAllInventorys', () => {
    expect(actions.selectAllCart()).toMatchSnapshot()
    expect(actions.selectAllInventory()).toMatchSnapshot()
  })
  it('selectStonkInInventory/selectStonkInCart action', () => {
    expect(actions.selectStonkInInventory(3)).toMatchSnapshot()
    expect(actions.selectStonkInCart(4)).toMatchSnapshot()
  })
  it('removeStonk action', () => {
    expect(actions.removeStonk(45)).toMatchSnapshot()
  })
  it('openSnackbar/closeSnackbar action', () => {
    expect(actions.openSnackbar('msg', 'success')).toMatchSnapshot()
    expect(actions.closeSnackbar()).toMatchSnapshot()
  })
})
