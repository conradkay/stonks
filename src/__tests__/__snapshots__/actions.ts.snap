// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`core actions add to cart action 1`] = `
Object {
  "amount": 2,
  "id": 3,
  "stonk": Object {
    "amount": 2,
    "description": "floopderop",
    "id": 1,
    "name": "floop",
    "pathToImage": "https://",
    "price": 101,
    "selected": false,
    "sellAmount": 1,
  },
  "type": "ADD_TO_CART",
}
`;

exports[`core actions buy stonks action 1`] = `
Object {
  "type": "BUY_STONKS",
}
`;

exports[`core actions change amount cart action 1`] = `
Object {
  "amount": 3,
  "index": 20,
  "type": "CHANGE_AMOUNT_CART",
}
`;

exports[`core actions change amount inventory action 1`] = `
Object {
  "amount": 4,
  "index": 40,
  "type": "CHANGE_AMOUNT_INVENTORY",
}
`;

exports[`core actions change amount shop action 1`] = `
Object {
  "amount": 3,
  "index": 2,
  "type": "CHANGE_AMOUNT_SHOP",
}
`;

exports[`core actions openSnackbar/closeSnackbar action 1`] = `
Object {
  "message": "msg",
  "type": "OPEN_SNACKBAR",
  "variant": "success",
}
`;

exports[`core actions openSnackbar/closeSnackbar action 2`] = `
Object {
  "type": "CLOSE_SNACKBAR",
}
`;

exports[`core actions removeStonk action 1`] = `
Object {
  "index": 45,
  "type": "REMOVE_STONK",
}
`;

exports[`core actions selectAllCart/selectAllInventorys 1`] = `
Object {
  "type": "SELECT_ALL_CART",
}
`;

exports[`core actions selectAllCart/selectAllInventorys 2`] = `
Object {
  "type": "SELECT_ALL_INVENTORY",
}
`;

exports[`core actions selectStonkInInventory/selectStonkInCart action 1`] = `
Object {
  "index": 3,
  "type": "SELECT_STONK_IN_INVENTORY",
}
`;

exports[`core actions selectStonkInInventory/selectStonkInCart action 2`] = `
Object {
  "index": 4,
  "type": "SELECT_STONK_IN_CART",
}
`;

exports[`core actions sell stonks action 1`] = `[Function]`;

exports[`core actions unSelectAllInventory/unSelectAllCart 1`] = `
Object {
  "type": "UNSELECT_ALL_INVENTORY",
}
`;

exports[`core actions unSelectAllInventory/unSelectAllCart 2`] = `
Object {
  "type": "SELECT_ALL_CART",
}
`;
